server/store/channel_activity.go

```diff
@@ -0,0 +1,124 @@
+package store
+
+import (
+	sq "github.com/Masterminds/squirrel"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+)
+
+const (
+	channelLimit = 1000
+	userLimit    = 1000
+)
+
+type userChannelActivity = map[string]map[string]int64 //map[userID]map[channelID]activity
+
+// GetChannelActivity returns user activity in most active channels for most active users.
+// For now user activity is the number of posts posted in a channel.
+func (db *Store) GetChannelActivity(teamID string) (userChannelActivity, error) {
+	channels, err := db.GetChannelsForTeam(teamID)
+	if err != nil {
+		return nil, err
+	}
+	users, err := db.getUsersForTeam(teamID)
+	if err != nil {
+		return nil, err
+	}
+	activity, err := db.getUserActivityForUsersInChannels(users, channels)
+	if err != nil {
+		return nil, err
+	}
+	return activity, nil
+}
+
+// GetChannelsForTeam returns most active channels in the team
+func (db *Store) GetChannelsForTeam(teamID string) ([]string, error) {
+	since := model.GetMillis() - period
+
+	query := db.sq.Select("C.Id").
+		From("Posts AS P").
```

Comment:
Joining on `Posts` might be painful. Is there another way we can estimate this data instead?